blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java

```diff
@@ -169,4 +194,35 @@ FlowNodeWrapper getFlowNodeWrapper(){
         return node;
     }
 
+    @Extension
+    public static class NodeFactoryImpl extends BlueTestResultFactory {
+        @Override
+        public Result getBlueTestResults(Run<?, ?> run, final Reachable parent) {
+            Iterable<BlueTestResult> results = null;
+            TestResultAction action = run.getAction(TestResultAction.class);
+            if (action != null && parent instanceof BluePipelineNode) {
```

Comment:
Other plugins are going to want to know how to do this kind of processing - why not encode this in the `BlueTestResultFactory`? 

The intent might be easier if we added something like: 
`public Result getBlueTestResults(Run<?,?> run, BluePipelineNode node, final Reachable parent)`

Then the code in this factory can move to that method on `io.jenkins.blueocean.service.embedded.rest.junit.BlueJUnitTestResult.FactoryImpl`

Reply:
Something like https://github.com/abayer/blueocean-plugin/pull/1